map/map_tf_generator/CMakeLists.txt

```diff
@@ -1,37 +1,49 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.5)
 project(map_tf_generator)
 
-add_compile_options(-std=c++14)
+### Compile options
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  roscpp
+### Dependencies
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(std_msgs REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(tf2_ros REQUIRED)
```

Comment:
Please use ament_cmake_auto to find dependencies

```
find_package(ament_cmake_auto REQUIRED)
ament_auto_find_build_dependencies()
```

Reply:
I tried once with ament_auto, but it didn't work for the PCL. @nnmm also said a similar problem. https://github.com/tier4/Pilot.Auto/pull/29#issuecomment-709397741

@mitsudome-r Any idea to solve this?

Reply:
This is what I tried.

```
cmake_minimum_required(VERSION 3.5)
project(map_tf_generator)

### Compile options
if(NOT CMAKE_CXX_STANDARD)
  set(CMAKE_CXX_STANDARD 14)
endif()
if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
  add_compile_options(-Wno-unused-parameter -Wall -Wextra -Wpedantic)
endif()

### Dependencies
find_package(ament_cmake_auto REQUIRED)
ament_auto_find_build_dependencies()

# Generate exe file
ament_auto_add_executable(map_tf_generator nodes/map_tf_generator.cpp)

## Install
ament_auto_package(INSTALL_TO_SHARE
  launch
)
```

Reply:
What I believe is that `ament_auto_find_build_dependencies()` only finds dependencies for colcon packages.
Therefore, you have to add dependency on non-ROS package(like PCL) yourself.
e.g. 
```
cmake_minimum_required(VERSION 3.5)
project(map_tf_generator)

### Compile options
if(NOT CMAKE_CXX_STANDARD)
  set(CMAKE_CXX_STANDARD 14)
endif()
if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
  add_compile_options(-Wno-unused-parameter -Wall -Wextra -Wpedantic)
endif()

### ROS Dependencies
find_package(ament_cmake_auto REQUIRED)
ament_auto_find_build_dependencies()

### non-ROS Dependencies
find_package(PCL REQUIRED)

# Generate exe file
ament_auto_add_executable(map_tf_generator nodes/map_tf_generator.cpp)
target_link_libraries(map_tf_generator ${PCL_LIBRARIES})

## Install
ament_auto_package(INSTALL_TO_SHARE
  launch
)

```

Reply:
This worked perfectly. Thanks!